### ISSUE TITLE
Interface for categorical parameters

### ISSUE THREAD

**Issue by fwuermseher** [E-MAB/GMAB#43]
## Issue 

Gmab samples integer decision parameters using the specified bounds. The package should also offer interfaces for users that want to run an optimization that has categorical parameters.

## Suggested Changes

- Add an interface that simplifies the process for setting CategoricalParams on top of the existing API (Inherits from BaseParam).

## Expected Outcome

- The User can define a categorical parameter based on a [Sequence](https://docs.python.org/3/library/collections.abc.html) of choices.
- Using indexation and the length of the Sequence, CategoricalParam creates integer bounds for gmab and maps internal representation to the real values.

```python
import gmab
param = gmab.CategoricalParam(['a', 'b', 'c'])

param.bounds # Output: [(0, 2)]

param.map_to_value([0]) # Output: 'a'
```

Note that, similar to other libaries, this implements the Categorical Parameter on an *ordinal scale*, which means the simulation results could be affected by the input order for the choices.

(see [optuna's Categorical Distribution](https://github.com/optuna/optuna/blob/master/optuna/distributions.py#L483) or [sklearn's label encoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html))
